packages/@extension-kid/toast-notifications/__tests__/ToastExtension-test.ts

```diff
@@ -0,0 +1,190 @@
+import ToasterExtension from "../ToastExtension";
+import { ToastCallbackType, ToastNotification } from "../ToastNotification";
+
+describe("ToasterExtension", () => {
+  describe("#supportedNotifications", () => {
+    it("returns toast symbol", () => {
+      const ext = new ToasterExtension();
+
+      expect(ext.supportedNotifications()).toEqual([
+        ToastNotification.NotificationType
+      ]);
+    });
+  });
+
+  describe("#push", () => {
+    it("updates stream when new notification is pushed", () => {
+      const ext = new ToasterExtension();
+      const notification = new ToastNotification("unit test");
+      const nextCallback = jest.fn();
+      ext.Toast$.subscribe(nextCallback);
+
+      ext.push(notification);
+
+      expect(nextCallback).toHaveBeenCalledTimes(2);
```

Comment:
i had a hard time understanding why this is `2` (as in "i naively did not expect it do be 2 and had to look up the implementation details").

could we `ext.push(notification);` before `ext.Toast$.subscribe(nextCallback);` so we only get one event? maybe this could avoid the need to do `nextCallback.mockReset();` in some of the subsequent tests.

Reply:
I think I want to refactor the creation of the extension and callback mock into `beforeEach` and then I can separate the two expected calls to two tests.